language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/dependencygraph/DependencyGraph.scala

```diff
@@ -5,39 +5,36 @@ package com.daml.lf.codegen.dependencygraph
 
 import com.daml.lf.codegen.Util
 import com.daml.lf.data.ImmArray.ImmArraySeq
-import com.daml.lf.codegen.lf.DefTemplateWithRecord
 import com.daml.lf.data.Ref.Identifier
-import com.daml.lf.iface.{InterfaceType, DefDataType}
-import scalaz.std.either._
+import com.daml.lf.iface.InterfaceType
 import scalaz.std.list._
 import scalaz.syntax.bifoldable._
-import scalaz.syntax.bifunctor._
 import scalaz.syntax.foldable._
 
 private[codegen] object DependencyGraph {
   def orderedDependencies(
       decls: Map[Identifier, InterfaceType]
-  ): OrderedDependencies[Identifier, Either[DefTemplateWithRecord, DefDataType.FWT]] = {
+  ): OrderedDependencies[Identifier, InterfaceType] = {
```

Comment:
`DefTemplateWithRecord` is a template definition alongside its associated record. An `InterfaceType` is either that or a `DefDataType`. Instead of this `Either`, I simply used the `InterfaceType`. We may want to use a different abstraction though, as I'll have to introduce the concept of interfaces at this level as part of fixing the thing I'm actually here for.

Reply:
`Either` will also help out because, just as `templateIds` and `typeDeclarations` were easily separated before by means of `partitionMap`, it may be more convenient to not have the LF interfaces in the same result value, thus easily separated out by means of `partitionMap`.

Reply:
Thanks!